swan-lake/learn-the-platform/source-code-dependencies/package-references.md

```diff
@@ -192,11 +182,69 @@ password = "<maven-repository-password>"
+When working with JAR file dependencies, it is best practice to attach them to the default root module of your package, if your package has only the default root module. This makes it easy to manage and maintain the dependencies. However, if your package is a Ballerina library package, it is recommended that you specify the JAR file dependencies in each Ballerina module that depends on the JAR file.
+
+By default, the "bal build" command will package all JAR files specified in the `Ballerina.toml` file along with the executable JAR file. This ensures that the executable JAR file includes all necessary dependencies, making the package self-contained and easy to use. This approach allows for a more organized and efficient way of managing dependencies and ensures that the 
+
+**Provide the path of JAR file**
+
+```toml
+[[platform.java11.dependency]]
+# Group ID of the dependency.
+groupId = "<group-id>"
+# Artifact ID of the dependency.
+artifactId = "<artifact-id>"
+# Version of the dependency.
+version = "<version>"
+# Absolute or relative path of the JAR file.
+path = "<path-to-jar-file-1>"
+```
+
+The Ballerina compiler will copy the specified JAR file from the provided path when creating the archive.
+
+* Restrict usage to specific modules.
+
+  ```toml
+  [[platform.java11.dependency]]
+  # Absolute or relative path of the JAR file.
+  path = "<path-to-jar-file-1>"
+  # An optional comma-separated list of Ballerina module names (to restrict the usage of this JAR).
+  modules = ["<ballerina-module-1>"]
+  ```
+
+  It is considered best practice to provide the names of the modules that use the JAVA library as a comma-separated list. This ensures that the JAR library is only used in the specified modules.
+
+* Resolve multiple versions of the same JAR
+
+  ```toml
+  [[platform.java11.dependency]]
+  # Absolute or relative path of the JAR file.
+  path = "<path-to-jar-file-1>"
+  # Optional details about the dependency (to handle conflicting JAR files).
+  groupId =  "<dependency-group-id>"
+  artifactId =  "<dependency-artifact-id>"
+  version =  "<dependency-version>"
+  ```
+
+  When a package has two dependencies that use different versions of the same Java library, the compiler picks the latest version. Two platform dependencies having the same `groupId` and the `artifactId` is considered to be the same where the latest out of the two will be picked for creating the executable JAR. 
+
+  For example, if you are using `commons-logging-1.2.jar` in your package but there is another package in the dependency graph that uses `commons-logging-1.1.1.jar`, the compiler will pick `commons-logging-1.2.jar` since it is the latest version. 
+
+  ```
+  WARNING [pkg3] detected conflicting jar files. 'native-1.0.1.jar' dependency of 'platformlib/pkg2' conflicts with 'native-1.0.0.jar'  dependency of 'platformlib/pkg1'. Picking 'native-1.0.1.jar' over 'native-1.0.0.jar'.
+  ```
+
+* Define the scope for a dependency
```

Comment:
```suggestion
* Define the scope for a dependency
```
Shall we make these sub-topics and add a one-line intro sentence as well?

Reply:
Added in [b6a5ef1](https://github.com/ballerina-platform/ballerina-dev-website/pull/6441/commits/b6a5ef1ab01b17dadaa434eb584fa1a665109a08)